syntax = 'proto3';

package o2.bookkeeping;

// Services

service FlpService {
  rpc CreateMany(ManyFlpsCreationRequest) returns (FlpList);
  rpc UpdateCounters(UpdateCountersRequest) returns (Flp);
}

// High level messages

message FlpCreationRequest {
  string name = 1;
  string hostname = 2;
  optional int32 runNumber = 3;
}

message ManyFlpsCreationRequest {
  repeated FlpCreationRequest flps = 1;
}

message UpdateCountersRequest  {
  string flpName = 1;
  int32 runNumber = 2;
  uint64 nSubTimeframes = 3;
  uint64 nEquipmentBytes = 4;
  uint64 nRecordingBytes = 5;
  uint64 nFairMQBytes = 6;
}

message FlpList {
  repeated Flp flps = 1;
}

// Low-level messages and enums

message Flp {
  int32 id = 1;
  string name = 2;
  string hostname = 3;
  // Unix timestamp when this entity was created.
  int64 createdAt = 4;
  // Unix timestamp when this entity was last updated.
  int64 updatedAt = 5;
  uint64 bytesEquipmentReadOut = 6;
  uint64 bytesFairMQReadOut = 7;
  uint64 bytesProcessed = 8;
  uint64 bytesRecordingReadOut = 9;
  uint64 nTimeframes = 10;
}
